Project/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs

```diff
@@ -33,9 +67,17 @@ public class GridAgent : Agent
 
     public override void Initialize()
     {
+        m_GoalSensor = this.GetComponent<VectorSensorComponent>();
         m_ResetParams = Academy.Instance.EnvironmentParameters;
     }
 
+    public override void CollectObservations(VectorSensor sensor)
+    {
+        Array values = Enum.GetValues(typeof(GridGoal));
```

Comment:
Can this happen somewhere else? It feels like abuse of CollectObservations(), since it's not touching the input VectorSensor.

Reply:
VectorSensor is null here, I do not see an issue with this. Goal Signal is an observation, so it makes sense to me that it is called in CollectObservation.
Would it be better if I put this logic into a `CollectGoal` method with no arguments that I call in CollectObservations ?

Reply:
CollectGoal is maybe for the example (but let's not add it Agent). Let me think about a better way.

One problem (which I didn't realize until now) is that we don't check for null CollectObservationsSensor during the normal update step:
https://github.com/Unity-Technologies/ml-agents/blob/e4e9c51bf1615953759249cc8ad349534d2569f4/com.unity.ml-agents/Runtime/Agent.cs#L1062
but we do check for null when the agent is done:
https://github.com/Unity-Technologies/ml-agents/blob/e4e9c51bf1615953759249cc8ad349534d2569f4/com.unity.ml-agents/Runtime/Agent.cs#L563-L571